### PR TITLE
bowtie2 added

### PR DESCRIPTION
**Pull request by HNH0303**
With samtools included to immediately convert sam files to bam files

### CODE REVIEW

**Comment by kapsakcj** [comment]
Hi Holly, thanks for the PR. This is an essential bioinfo tool that is missing from our collection.

I was able to build the docker image successfully and was able to align some single end sars-cov-2 reads to it's assembly, followed by `samtools view [...]` & `samtools sort [...]` which ran as expected. Everything checks out on my end ✔️ 

I have some minor suggestions for the dockerfile that I would like to put forward before we merge the PR and build the image on dockerhub and quay. I'll make comments on the dockerfile & README.md

File: README.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ For many people Docker is not an option, but Singularity is. Most Docker contain
...
 | [bcftools](https://hub.docker.com/r/staphb/bcftools/) <br/> [![docker pulls](https://img.shields.io/docker/pulls/staphb/bcftools.svg?style=popout)](https://hub.docker.com/r/staphb/bcftools) | <ul><li>1.10.2</li><li>1.11</li><li>1.12</li></ul> | https://github.com/samtools/bcftools |
 | [bedtools](https://hub.docker.com/r/staphb/bedtools/) <br/> [![docker pulls](https://img.shields.io/docker/pulls/staphb/bedtools.svg?style=popout)](https://hub.docker.com/r/staphb/bedtools) | <ul><li>2.29.2</li><li>2.30.0</li></ul> | https://bedtools.readthedocs.io/en/latest/ <br/>https://github.com/arq5x/bedtools2 |
+| [bowtie2](https://hub.docker.com/r/staphb/bowtie2/) <br/> [![docker pulls](https://img.shields.io/docker/pulls/staphb/bowtie2.svg?style=popout)](https://hub.docker.com/r/staphb/bowtie2) | <ul><li>2.29.2</li><li>2.30.0</li></ul> | http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml <br/>https://github.com/BenLangmead/bowtie2 |
```
Comment (kapsakcj) [inline README.md:29]: The middle column of the table should be: `<ul><li>2.4.4</li></ul>`  To match the version of bowtie2 included in the docker image

File: bowtie2/2.4.4/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# FROM defines the base docker image to start from. This command has to come first in the file
+FROM ubuntu:xenial
+
+ARG SAMTOOLSVER=1.11
+ARG BOWTIE2VER=2.4.4
+
+# metadata (there are a few other labels you can add, these are optional but preferred!)
+LABEL base.image="ubuntu:xenial"
+LABEL version="1"
```
Comment (kapsakcj) [inline bowtie2/2.4.4/Dockerfile:9]: please change this line `LABEL version="1"` --> `LABEL dockerfile.version="1"` for clarity. See explanation here: https://github.com/StaPH-B/docker-builds/issues/118

File: bowtie2/2.4.4/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+LABEL software.versions="2.4.4,\n 1.10"
+LABEL description="Bowtie2: Genome assembler using a reference and mapping\n Samtools: a set of tools for interacting with and reformatting sequence data"
+LABEL website="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml\n https://www.htslib.org/"
+LABEL maintainer="Holly Halstead"
+LABEL maintainer.email="holly.halstead@doh.wa.gov"
+
+ENV ZIP=bowtie2-${BOWTIE2VER}-linux-x86_64.zip
+ENV URL=https://github.com/BenLangmead/bowtie2/releases/download/v${BOWTIE2VER}/
+ENV FOLDER=bowtie2-${BOWTIE2VER}
+ENV DST=/home/dock/bin
```
Comment (kapsakcj) [inline bowtie2/2.4.4/Dockerfile:20]: I don't believe the $DST directory is used anywhere? Might be good to remove this line as well as the command below on line 24 to make the directory `mkdir -p $DST && \`

Doesn't hurt to keep it around though. simply an empty directory

Comment (kapsakcj) [reply bowtie2/2.4.4/Dockerfile:20]: Looks like all of these variables are unused, at least in the docker image building process. Maybe I'm missing the reasoning these are included

Up to you if you want to remove them!

**Comment by kapsakcj** [comment]
OK, I made my comments. Please let me know if you have any questions or if I can clarify anything.

Thank you!

**Comment by HNH0303** [comment]
Changes were made as suggested and tested locally. Thanks for your feedback!

**Comment by kapsakcj** [comment]
Awesome, thank you. Tested out great on my end again. I will merge and push to dockerhub and quay.

**Comment by kapsakcj** [comment]
The docker image is now available on dockerhub and quay. Thank you for the PR!

https://hub.docker.com/r/staphb/bowtie2/tags?page=1&ordering=last_updated

https://quay.io/repository/staphb/bowtie2?tab=tags

**Pull request merged**